fabric-data-attachment-api-v1/src/main/java/net/fabricmc/fabric/api/attachment/v1/AttachmentType.java

```diff
@@ -66,13 +65,40 @@ default boolean isPersistent() {
 	 * As an example, for a (mutable) list/array attachment type,
 	 * the initializer should create a new independent instance each time it is called.</p>
 	 *
-	 * @return the initializer for this attachment
+	 * @return the initializer for this attachment, may be {@code null}
 	 */
 	@Nullable
 	Supplier<A> initializer();
 
+	/**
+	 * @return the {@link EntityCopyHandler} of this attachment type, may be {@code null}
+	 */
+	@Nullable
+	AttachmentType.EntityCopyHandler<A> entityCopyHandler();
+
 	/**
 	 * @return whether the attachments should persist after a player's death and respawn
 	 */
-	boolean copyOnPlayerRespawn();
+	boolean copyOnDeath();
+
+	/**
+	 * A functional interface to handle copying attachment data from an old entity instance to a new one, for example
+	 * during player respawn, entity conversion, or when an entity is teleported between worlds.
+	 *
+	 * <p>It is <i>imperative</i> that the data attached to the new entity doesn't hold a reference to the old entity,
+	 * as that can and will break in unexpected ways.</p>
+	 */
+	@FunctionalInterface
+	interface EntityCopyHandler<T> {
```

Comment:
I dont think this is needed, the data should not be holding onto refrences beyond what it plans to save. Lets discuss this before you revert it.